mirgecom/euler.py

```diff
@@ -98,51 +106,91 @@ def euler_operator(discr, state, gas_model, boundaries, time=0.0,
-    interior_states = make_fluid_state_trace_pairs(interior_cv, gas_model,
+    dd_base = as_dofdesc("vol")
+    dd_vol = DOFDesc("vol", quadrature_tag)
+    dd_faces = DOFDesc("all_faces", quadrature_tag)
+
+    def interp_to_surf_quad(utpair):
+        local_dd = utpair.dd
+        local_dd_quad = local_dd.with_discr_tag(quadrature_tag)
+        return TracePair(
+            local_dd_quad,
+            interior=op.project(discr, local_dd, local_dd_quad, utpair.int),
+            exterior=op.project(discr, local_dd, local_dd_quad, utpair.ext)
+        )
+
+    boundary_states = {
+        btag: project_fluid_state(
+            discr, dd_base,
+            # Make sure we get the state on the quadrature grid
+            # restricted to the tag *btag*
+            as_dofdesc(btag).with_discr_tag(quadrature_tag),
+            state, gas_model) for btag in boundaries
+    }
+
+    cv_interior_pairs = [
+        # Get the interior trace pairs onto the surface quadrature
+        # discretization (if any)
+        interp_to_surf_quad(tpair)
+        for tpair in interior_trace_pairs(discr, state.cv)
+    ]
+
+    tseed_interior_pairs = None
+    if state.is_mixture > 0:
+        # If this is a mixture, we need to exchange the temperature field because
+        # mixture pressure (used in the inviscid flux calculations) depends on
+        # temperature and we need to seed the temperature calculation for the
+        # (+) part of the partition boundary with the remote temperature data.
+        tseed_interior_pairs = [
+            # Get the interior trace pairs onto the surface quadrature
+            # discretization (if any)
+            interp_to_surf_quad(tpair)
+            for tpair in interior_trace_pairs(discr, state.temperature)
+        ]
+
+    interior_states = make_fluid_state_trace_pairs(cv_interior_pairs,
+                                                   gas_model,
                                                    tseed_interior_pairs)
 
-    # Compute volume contributions
-    inviscid_flux_vol = inviscid_flux(state)
-    # Compute interface contributions
+    # Interpolate the fluid state to the volume quadrature grid
+    # (conserved vars and derived vars if applicable)
+    state_quad = project_fluid_state(discr, dd_base, dd_vol, state, gas_model)
+
+    inviscid_flux_vol = inviscid_flux(state_quad)
     inviscid_flux_bnd = (
-        # Interior faces
-        sum(inviscid_facial_flux(discr, gas_model, pair,
-                                 numerical_flux_func=inviscid_numerical_flux_func)
-            for pair in interior_states)
-        # Domain boundary faces
-        + sum(
-            boundaries[btag].inviscid_divergence_flux(
-                discr, btag, gas_model, state_minus=boundary_states[btag], time=time,
-                numerical_flux_func=inviscid_numerical_flux_func)
+
+        # Domain boundaries
+        sum(boundaries[btag].inviscid_divergence_flux(
```

Comment:
Fix the indentation to make the body of the comprehension visually distinguishable.